pkg/reconciler/ingress/controller.go

```diff
@@ -120,6 +123,12 @@ func NewController(
 		DeleteFunc: statusProber.CancelPodProbing,
 	})
 
+	labelProvider := NewLabelProvider(c.Logger.Named("label-provider"), kubeclient.Get(ctx))
+	deploymentInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		// Add istio ingress labels
+		UpdateFunc: labelProvider.process,
```

Comment:
This will operate on all deployments in the cluster which is not what you want. You want only the deployments owned by Revisions with the ingress class that is being used.